pypyr/dsl.py

```diff
@@ -562,12 +562,14 @@ def run_conditional_decorators(self, context):
         # change whether they evaluate True or False, thus apply formatting at
         # last possible instant.
         run_me = context.get_formatted_as_type(self.run_me, out_type=bool)
-        skip_me = context.get_formatted_as_type(self.skip_me, out_type=bool)
-        swallow_me = context.get_formatted_as_type(self.swallow_me,
-                                                   out_type=bool)
 
         if run_me:
+            skip_me = context.get_formatted_as_type(self.skip_me,
+                                                    out_type=bool)
+
             if not skip_me:
+                swallow_me = context.get_formatted_as_type(self.swallow_me,
```

Comment:
if the idea is _only_ to evaluate this if strictly necessary from a control-of-flow perspective, putting this here still means `swallow_me` evaluates regardless of whether of the step actually hits an exceptions and _needs_ to check it?

i.e if this line moves to just before `if swallow_me:` in the try/catch handler then it'll evaluate just in time, and also raise an error that'll break out of the exception handler with the formatting error for the pipeline author to see, while still saving the original pipeline run error to the errors list.

or am I missing something?

Reply:
Good suggestion, thank you. I have added some extra code to deal with the swallow parsing error